### DISCUSSION TITLE
ControlNet API: what is the "mask" field in control net unit JSON?

### DISCUSSION THREAD

**Discussion by thundercarrot** [Mikubill/sd-webui-controlnet#434] [Q&A]
Does it have an analogue in the UI? I'm not seeing it.
Thanks!

**Comment by sraney** [comment]
Yeah, it's in the Automatic1111 script dropdown.  <a href="https://ibb.co/bQ44JhF"><img src="https://i.ibb.co/xXnnHvY/Screenshot-2023-02-28-210829.png" alt="Screenshot-2023-02-28-210829" border="0"></a>)

**Reply by thundercarrot** [reply]
The m2m script looks like a way to batch process frames from a movie to create a new movie? What does that have to do with the mask field in the API?

**Reply by thundercarrot** [reply]
What I'm really doing (which I also asked on the ControlNet github discussion): I'm supplying my own depth image. I also have a mask that indicates where depth image pixel values are invalid. Setting invalid depth image pixels to 0 seems wrong, as that would mean "very far away".

How can I use this mask with the depth image model?

**Reply by kft334** [reply]
Still no answer? I would like to know what those masks are for too. When I use a mask in a ControlNetUnit it appears that the mask is used as the input to be processed rather than the control input image.

**Reply by physis123** [reply]
Did anyone figure this out? I can't work out what the mask image does when passed in to the API

**Reply by ljleb** [reply]
`mask` is the mask for the input image to controlnet. It is ignored at the moment in api when no image is passed at the same time, even when falling back on `p.init_images[0]`. (the img2img image) I have not tested it, all I know is that it exactly corresponds to what is inpainted in the gradio control unit image components.
